### PR TITLE
3D Box limiters example: small edit in velocity for 3rd-order upwinding

### PR DESCRIPTION
**Pull request by valeriabarra**
@OsKnoth suggested that we use the velocity interpolated at faces (both horizontal and vertical velocity components) in the third-order upwinding operator. 

- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

**Comment by valeriabarra** [comment]
bors try

File: examples/hybrid/box/limiters_advection.jl [hunk 1]
```diff
@@ -262,11 +262,15 @@ for (k, horz_ne) in enumerate(horz_ne_seq)
...
+        fuₕ = first_order_Ic2f.(cuₕ)
+        fuvw =
+            Geometry.Contravariant123Vector.(fuₕ) .+
+            Geometry.Contravariant123Vector.(fw)
+        vert_flux_wρ = vdivf2c.(fw .* first_order_Ic2f.(y.ρ))
         vert_flux_wρq =
             vdivf2c.(
                 first_order_Ic2f.(y.ρ) .*
-                third_order_upwind_c2f.(w, y.ρq ./ y.ρ),
+                third_order_upwind_c2f.(fuvw, y.ρq ./ y.ρ),
```
Comment (valeriabarra) [inline examples/hybrid/box/limiters_advection.jl:273]: @OsKnoth here is where we use `fuvw`

**Comment by simonbyrne** [comment]
Does this make a difference? It will in the presence of topography, but shouldn't otherwise?

**Comment by OsKnoth** [comment]
without orography the first term do nothing, but with orography its different and that is our goal.

**Comment by valeriabarra** [comment]
> Does this make a difference? It will in the presence of topography, but shouldn't otherwise?

The results here in this branch compared to `main` are indeed identical. Do we still want to merge these changes in preparation of what we will need with topography? 

Also, do you know by any chance why buildkite says it cannot find the artifacts? In the error message, I see that the prefix `examples/` from the path is missing, but the file exists and has the full file name, including the `examples/` bit, so I'm not sure what is going on. Thanks!

**Comment by valeriabarra** [comment]
@simonbyrne the results here are identical as expected, since there is no topography for now. Do we want to go ahead and merge this and then I can propagate this same change both in #714  (if we want to update that example without limiters) and #702 ? 

Did you have a chance to see my question above? 

> Also, do you know by any chance why buildkite says it cannot find the artifacts? In the error message, I see that the prefix `examples/` from the path is missing, but the file exists and has the full file name, including the `examples/` bit, so I'm not sure what is going on. Thanks!

**Pull request merged**